[lib/gen]: fixed __int128 parsing

lukeg101 opened the pull request:
Motivation: Test int128 programs generated by diy are not parsed by herd.

This patch fixes the following bugs in the int128 implementation:
- `diy7` did not parse `-type uint128_t`
- litmus tests with the initial state `_Atomic __int128 x = 0` did not parse
- `diy` generates C/C++ litmus tests using the `(u)int128_t` which not valid in LLVM/GCC. Instead we generate tests using `__(u)int128`.

A future patch will fix `herd` so that it accepts `__int128` data in AArch64 (It is conceptually a separate patch that deals with Semantics).

Tests are provided in this patch

lukeg101 commented:
This patch is ready for review and will help colleagues with a compiler question.

maranget commented on herd/tests/instructions/C/C09.litmus:
This test is the same as C08.litmus... Delete?

maranget commented on herd/tests/instructions/C/C10.litmus:
```diff
@@ -0,0 +1,19 @@
+C LB004
```
```suggestion
C C10
```

maranget commented on herd/tests/instructions/C/C11.litmus:
```diff
@@ -0,0 +1,19 @@
+C LB004
```
```suggestion
C C11
```

maranget commented on herd/tests/instructions/C/C10.litmus:
For what it's worth, my gcc does not know about the `__uint128_t` type

maranget commented on herd/tests/instructions/C/C11.litmus:
For what it's worth, my version of gcc chokes on this test, with message:
```
undefined reference to __atomic_load_16
```

maranget replied:
My remark was.. a remark. Given the experimental nature of the 128 bit integer types for herdtools7 and the work involved in a complete implementation, we can accept such limitations.

maranget reviewed (changes requested):
Before merging I would like the tests to be given internal names that correspond to their file name. Moreover test C09 is a duplicate of test C08.

lukeg101 commented:
Hi Luc,
Thank you for this code review, I appreciate it. I have deleted the redundant test and renamed the others to follow the test convention.

A note on `atomic_load_16`, GCC and LLVM differ on the code generation for 128-bit atomic loads, precisely LLVM emits inlined assembly and GCC emits this function call. To ensure your GCC works you need to link in the correct atomics library, for instance https://godbolt.org/z/KEf1ncEMK.

As far as I understand we do not support function calls of this kind in the tool suite. Shall I remove C11.litmus?

maranget reviewed (approved):
All changes performed, thanks.

maranget commented:
Squashed and merged. Thanks @lukeg101.
